src/Controls/src/Core/BindableLayout/BindableLayout.cs

```diff
@@ -398,14 +493,37 @@ View CreateEmptyView(object emptyView, DataTemplate dataTemplate)
 		}
 
 		void ItemsSourceCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
+		{
+			if (_dispatcher?.IsDispatchRequired ?? false)
+			{
+				_dispatcher.Dispatch(() => UpdateCollection(e));
```

Comment:
For now, let's remove the dispatcher part of this PR and discuss in a separate PR and Issue. There's a lot of magic in this PR that's going to get delayed discussing dispatcher changes. This also would probably be more of a net9 change.

Reply:
I've just removed the dispatcher logic in a separate commit which you/I can easily revert in the future when the time comes :)